back/times/tasks.py

```diff
@@ -42,12 +42,15 @@ def process_log_times(end_date=None, day_window=None, run_code=None, course=None
     """
     def make_row(row, date, course):
         delta = row["delta_time"].days*84600 + row["delta_time"].seconds
+        vertical_to_associate = CourseVertical.objects.filter(
```

Comment:
This step will call the DB on each step. A local cache should be implemented to avoid it. Like a dictionnary.